server/testrun/testrun.go

```diff
@@ -347,42 +353,61 @@ finishedloop:
 
 }
 
-func cleanTestData(dirtyData map[string]string) map[string]map[string]map[string]float32 {
+// cleanTestData cleans up the testing data that comes back from the agents.
+// The test data that comes back raw from the agents is "dirty", meaning it is designed to be as flexible
+// as possible.
+func cleanTestData(dirtyData map[string]string) (map[string]map[string]map[string]interface{}, error) {
```

Comment:
Very good points - I should have delved into the `json` package more deeply. I implemented these fixes and am tracking down a related issue, will post again when I have a solid commit together